kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/ScoreWrapper.java

```diff
@@ -0,0 +1,68 @@
+package org.kie.server.api.model.instance;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlValue;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamConverter;
+import com.thoughtworks.xstream.converters.extended.ToAttributedValueConverter;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.impl.score.ScoreUtils;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XStreamConverter(value = ToAttributedValueConverter.class, strings = { "scoreString" })
+public class ScoreWrapper {
+
+    @XmlAttribute(name = "scoreClass")
+    @XStreamAlias(value = "scoreClass")
+    private Class<? extends Score> scoreClass;
+
+    @XmlValue
+    private String scoreString;
+
+    // Define default constructor to enable class marshalling/unmarshalling
+    private ScoreWrapper() {
+    }
+
+    public ScoreWrapper( Score score ) {
+        if ( score == null ) {
+            throw new IllegalArgumentException( "The score parameter cannot be null." );
+        }
+        this.scoreClass = score.getClass();
+        this.scoreString = score.toString();
+    }
+
+    public Class<? extends Score> getScoreClass() {
+        return scoreClass;
+    }
+
+    public String getScoreString() {
+        return scoreString;
+    }
+
+    /**
+     * Returns score representation of the object.
+     *
+     * @return Score representation of the object.
+     * @throws UnsupportedOperationException If <code>scoreClass</code> is not one of the out-of-box score implementations. In this case
+     *                                       clients may implement their own way to extract the score object.
+     */
+    public Score toScore() {
+        if ( scoreClass == null ) {
+            return null;
```

Comment:
This should fail fast, because the constructor guarantees that scoreClass is never null.

So when it is null, that means JAXB or XStream messed up, and will want to fail fast here, as soon as possible. Otherwise this null will leak and potentially end up as garbage in the user's database for months in production before they notice this bug.